business/openid_auth.go

```diff
@@ -0,0 +1,140 @@
+package business
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+)
+
+type OpenIdMetadata struct {
+	// Taken from https://github.com/coreos/go-oidc/blob/8d771559cf6e5111c9b9159810d0e4538e7cdc82/oidc.go
```

Comment:
Isn't there a library we can pull in for this structure? What if something changes in the future? We would need to keep up-to-date with this metadata structure and manually update our code (rather than just updating a library).